lib/email/receiver.rb

```diff
@@ -660,102 +660,97 @@ def trim_discourse_markers(reply)
       reply.split(reply_above_line_regex)[0]
     end
 
-    def parse_from_field(mail = nil)
-      mail ||= @mail
-
+    def parse_from_field(mail, process_forwarded_emails: true)
       if email_log.present?
         email = email_log.to_address || email_log.user&.email
         return email, email_log.user&.username
       elsif mail.bounced?
-        Array.wrap(mail.final_recipient).each { |from| return extract_from_address_and_name(from) }
+        # "Final-Recipient" has a specific format (<name> ; <address>)
```

Comment:
This is a weird one, the `Final-Recipient` has a very specific format. It's not even a "standard" email addresses header. So it can't be parsed with the same method.